newIDE/app/src/ObjectsList/index.js

```diff
@@ -418,205 +611,235 @@ const ObjectsList = React.forwardRef<Props, ObjectsListInterface>(
-          if (!doRename) return;
+        const validatedNewName = objectFolderOrObject.isFolder()
+          ? newName
+          : getValidatedObjectOrGroupName(newName, global);
+        onRenameObjectFolderOrObjectWithContextFinish(
+          item,
+          validatedNewName,
+          doRename => {
+            if (!doRename) return;
 
-          onObjectModified(false);
-        });
+            onObjectModified(false);
+          }
+        );
       },
       [
         getValidatedObjectOrGroupName,
         onObjectModified,
-        onRenameObjectStart,
-        onRenameObjectFinish,
+        onRenameObjectFolderOrObjectWithContextFinish,
       ]
     );
 
-    const scrollToItem = (objectWithContext: ObjectWithContext) => {
-      if (sortableList.current) {
-        sortableList.current.scrollToItem(objectWithContext);
+    const editItem = React.useCallback(
+      (item: TreeViewItem) => {
+        if (item.isRoot || item.isPlaceholder) return;
+        const { objectFolderOrObject } = item;
+        if (objectFolderOrObject.isFolder()) return;
+
+        onEditObject(objectFolderOrObject.getObject());
+      },
+      [onEditObject]
+    );
+
+    const scrollToItem = (
+      objectFolderOrObjectWithContext: ObjectFolderOrObjectWithContext
+    ) => {
+      if (treeViewRef.current) {
+        treeViewRef.current.scrollToItem(objectFolderOrObjectWithContext);
       }
     };
 
-    const lists = enumerateObjects(project, objectsContainer);
-    const displayedObjectWithContextsList = filterObjectsList(
-      lists.allObjectsList,
-      {
-        searchText,
-        selectedTags: selectedObjectTags,
-      }
-    );
-    const selectedObjects = displayedObjectWithContextsList.filter(
-      objectWithContext =>
-        selectedObjectNames.indexOf(objectWithContext.object.getName()) !== -1
-    );
-    const displayedRenamedObjectWithContext = displayedObjectWithContextsList.find(
-      isSameObjectWithContext(renamedObjectWithContext)
+    const {
+      projectObjectFolderOrObjectsList,
+      containerObjectFolderOrObjectsList,
+    } = enumerateObjectFolderOrObjects(project, objectsContainer);
+    const getTreeViewData = React.useCallback(
+      (i18n: I18nType): Array<TreeViewItem> => {
+        const treeViewItems = [
+          {
+            label: i18n._(t`Global Objects`),
+            children:
+              projectObjectFolderOrObjectsList.length > 0
+                ? projectObjectFolderOrObjectsList
+                : [
+                    {
+                      label: (
+                        <Trans>
+                          There is no{' '}
+                          <Link
+                            href={globalObjectsWikiLink}
+                            onClick={() =>
+                              Window.openExternalURL(globalObjectsWikiLink)
+                            }
+                          >
+                            global object
```

Comment:
will this get translated well? 🤔

Reply:
Truth be told, I'm not sure, I'll see if we did it somewhere else (adding a link to a translated copy)

Reply:
You could consider using markdown + a placeholder for the link `There is no [globla object](LINK_PLACEHOLDER_DO_NOT_TRANSLATE_THIS) yet.`

I think we did something like this, but it's a bit hacky/fragile.

Reply:
We already used the syntax `<Trans>...<Link>...</Link>...</Trans>` here: https://github.com/4ian/GDevelop/blob/master/newIDE/app/src/MainFrame/Preferences/PreferencesDialog.js#L151-L166

And it seems to work fine!